### PR TITLE
[AU4] added view for Amplify effect

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

**Review by saintmatthieu — APPROVED** [review]
A few comments that are easy to address, so approving already. Thank you!

File: au4/src/effects/effects_base/view/effectsuiengine.cpp [hunk 1]
```diff
@@ -24,12 +24,17 @@ void EffectsUiEngine::setup()
 
 QQmlEngine* EffectsUiEngine::engine()
 {
-    if (!m_engine) {
-        m_engine = new QQmlEngine(this);
-        setup();
-    }
+    //! NOTE At the moment we are returning the main one.
+    //! We are making Qml view for built-in effects
+    //! It is not yet clear whether Qml is needed for custom ones
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply au4/src/effects/effects_base/view/effectsuiengine.cpp:29]: We assumed that we would provide an infrastructure for creating UI custom external effects (plugins). If we decide to do this, we should use a separate Qml engine, not the main one. To be able to know what we provide (API) and should keep stable, and what we do not and can easily change without worrying about backward compatibility

File: au4/src/effects/effects_base/internal/effectsprovider.cpp [hunk 2]
```diff
@@ -109,33 +109,16 @@ muse::Ret EffectsProvider::showEffect(const muse::String& type, const EffectInst
     return rv.ret;
 }
 
-muse::Ret EffectsProvider::performEffect(AudacityProject& project, Effect* effect, EffectSettings& settings,
-                                         const EffectTimeParams& timeParams)
+muse::Ret EffectsProvider::performEffect(AudacityProject& project, Effect* effect, std::shared_ptr<EffectInstance> pInstanceEx,
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply au4/src/effects/effects_base/internal/effectsprovider.cpp:112]:
```
returnVal = pInstanceEx->Process(settings);
```

File: au4/src/effects/effects_base/qml/Audacity/Effects/EffectsViewer.qml [hunk 3]
```diff
@@ -12,11 +12,13 @@ Rectangle {
     property string type: ""
     property string instanceId: ""
 
+    property string title: builder.contentItem ? builder.contentItem.title : ""
+    property bool isAllowApply: builder.contentItem ? builder.contentItem.isAllowApply : false
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply au4/src/effects/effects_base/qml/Audacity/Effects/EffectsViewer.qml]: fixed

File: au4/src/effects/builtin/amplify/amplifyviewmodel.cpp [hunk 4]
```diff
@@ -3,35 +3,171 @@
...
 void AmplifyViewModel::init()
 {
-    m_effect = dynamic_cast<AmplifyEffect*>(this->effect());
-    IF_ASSERT_FAILED(m_effect) {
+    AmplifyEffect* ae = effect();
+    IF_ASSERT_FAILED(ae) {
+        return;
+    }
+
+    double initAmp = ae->defaultAmp();
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply au4/src/effects/builtin/amplify/amplifyviewmodel.cpp]: fixed

File: au4/src/effects/builtin/amplify/amplifyviewmodel.cpp [hunk 5]
```diff
@@ -3,35 +3,171 @@
...
+    ae->setAmp(initAmp);
+    m_amp.val = std::numeric_limits<double>::lowest();
+
+    m_cliping = ae->isCanClip();
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply au4/src/effects/builtin/amplify/amplifyviewmodel.cpp]: fixed

**Pull request merged**